src/google_cloud_debugger/google_cloud_debugger_lib/dbg_breakpoint.cc

```diff
@@ -213,9 +258,47 @@ HRESULT DbgBreakpoint::PopulateBreakpoint(Breakpoint *breakpoint,
   location->set_line(line_);
   location->set_path(file_name_);
 
+  if (!expressions_map_.empty()) {
+    HRESULT hr = PopulateExpression(breakpoint, eval_coordinator);
+    if (FAILED(hr)) {
+      return hr;
+    }
+  }
+
   return stack_frames->PopulateStackFrames(breakpoint, eval_coordinator);
 }
 
+HRESULT DbgBreakpoint::PopulateExpression(
+    Breakpoint *breakpoint,
+    IEvalCoordinator *eval_coordinator) {
+  std::queue<VariableWrapper> bfs_queue;
+
+  for (auto &&kvp : expressions_map_) {
+    Variable *expression_proto = breakpoint->add_evaluated_expressions();
+
+    expression_proto->set_name(kvp.first);
+    const std::shared_ptr<DbgObject> &expression_value = kvp.second;
+    if (!expression_value) {
+      continue;
+    }
+
+    bfs_queue.push(VariableWrapper(expression_proto, expression_value));
+  }
+
+  if (bfs_queue.size() != 0) {
+    int half_available_breakpoint_size =
```

Comment:
The stackdriver debugger team suggests not having a limit on expressions.  This allows users to be able to get more data in cases where we may have truncated it otherwise.